src/classes/CMT_MetadataAPI.cls

```diff
@@ -177,7 +185,30 @@ public class CMT_MetadataAPI {
 
             if (isSuccess) {
                 if (!Test.isRunningTest()) {
-                    UTIL_OrgTelemetry_SVC.asyncProcessNpspSettingsTelemetry();
+                    Set<String> metadataObjectTypeSet = new Set<String>();
```

Comment:
Hey @patrick-yan-sf This is the start of the changes we discussed.

Reply:
@KietTruongSFDO - yup that's the general idea.  Here are some thoughts.

Do you think the metadataFullName.split call might have issues with namespaces?  I can't recall if the fullname will be npsp.opp_state_to_stage_mapping__mdt or just npsp__opp_state_to_stage_mapping__mdt.

I'm not sure about passing in the telemetry code version of the names.  That would couple this logic with that class.  Can we just pass in the full name and have the UTIL_OrgTelemetry_SVC class do that parsing?

Also let's move this logic into a separate function.

Something like 
Set<String> buildChangedMetada(DeployResult results) 

Then the call can just be Util_OrgTelemetry_SVC.asyncProcessCMTChange(buildChangedMetadata(results));